app/models/endpoint.rb

```diff
@@ -11,4 +11,24 @@ class Endpoint < ApplicationRecord
   validates :endpoint_node, presence: true
   validates :storage_location, presence: true
   validates :recovery_cost, presence: true
+
+  # rubocop:disable Metrics/AbcSize
+  # TODO: find a clean way to satisfy this style check, everything that silenced the complaint for me was
+  # less readable to my eyes due to what felt like needless indirection.
+  def self.seed_local_online_endpoints_from_config(preservation_policies = nil)
+    default_endpoint_type_config = Settings.endpoint_types.first[1]
+    endpoint_defaults = default_endpoint_type_config.endpoint_defaults
+    endpoint_type = EndpointType.default_local_online_endpoint_type
+    preservation_policies = [PreservationPolicy.default_preservation_policy]
+    Settings.moab.storage_roots.map do |storage_root_name, storage_root_location|
+      find_or_create_by!(endpoint_name: storage_root_name.to_s) do |endpoint|
+        endpoint.endpoint_type = endpoint_type
```

Comment:
any reason not to put the right side of line 21  on the right side of line 25?   I would hope there is an optimizing compiler that would only do the lookup once, and i think it might make the code easier to read.  I'm having trouble tracking all the variables set above.

Reply:
well, `EndpointType.default_local_online_endpoint_type` does a database lookup, and so it's something i'd like to not do every loop iteration, but also something i don't feel comfortable caching for more than the duration of this method call.  what about a comment indicating the reasoning for this approach?

Reply:
ah - yeah, a comment.